lib/src/features/home/app_list_provider.dart

```diff
@@ -0,0 +1,32 @@
+import 'package:app_buy_sell/src/features/home/application/app_service.dart';
+import 'package:app_buy_sell/src/features/home/domain/app_model.dart';
+import 'package:riverpod_annotation/riverpod_annotation.dart';
+
+part 'app_list_provider.g.dart';
+
+@riverpod
+class AppList extends _$AppList {
+  final AppService _appService = AppService();
+
+  @override
+  FutureOr<List<AppModel>> build() async {
+    return _initData();
+  }
+
+  Future<List<AppModel>> _initData() async {
+    final response = await _appService.getApps();
+    final data = response.$1;
+    return data;
+  }
+
+  Future<void> getApps() async {
+    state = const AsyncValue.loading();
+    try {
+      final response = await _appService.getApps();
+      final data = response.$1;
+      state = AsyncValue.data(data);
+    } catch (err, stack) {
+      state = AsyncValue.error(err, stack);
+    }
+  }
```

Comment:
Setting loading and error states is not necessary.

Reply:
@mafreud You don't show loading and errors ?
 for this screen or all?

Reply:
Riverpod handles loading and error state.

Reply:
@mafreud need set state for river pod https://riverpod.dev/docs/providers/notifier_provider
![2](https://github.com/ailab-dev/app_buy_sell/assets/117698659/0d28dfd2-5bb8-4c86-a5bb-8f30227b6cfc)

Reply:
Setting AsyncValue.data state is enough, I think. Could you try this?

Reply:
@mafreud ok, I fixed